### DISCUSSION TITLE
Using HTML template within JS

### DISCUSSION THREAD

**Discussion by LavissaWoW** [inventree/InvenTree#4647] [General]
Working on #4606, I feel it's wise to apply the HTML in clip.html to implement a copy-to-clipboard button inside tables rather than reinventing the wheel. However, it's become apparent that I can't use the {% include %} tag in a .js-file, as that alters the MIME-type with the browser subsequently refusing to load the offending file.

Without a long-standing knowledge of the Django template framework, I can't seem to identify a solution that allows me to render this HTML dynamically inside a JS function other than manually copying it in. I don't really feel that's a very elegant solution, nor is it very future oriented.

Basically, doing this:
`` return `${renderLink(somedata)}{% include clip.html %}` ``
turns the .js file into MIME type text/html. 

Am I doomed to write out the HTML in a separate function, or is there a way I can utilise clip.html from js?

**Comment by SchrodingersGat** [comment]
I think the cleaner solution would be to write a wrapper function in JS, much like `renderLink` and then wrap the content in that function. 

e.g.

```javascript
clipboard(renderLink(value));
```

Or something to that effect

**Reply by LavissaWoW** [reply]
That was my immediate idea. However, this will break when the project transitions into a React-based frontend. And I'm not very fond of code repetition if it can be avoided.

But as renderLink's return is written in pure HTML, I suppose that creating a similar helper in helpers.js won't add too much extra overhead when converting from bootstrap to React.

**Comment by LavissaWoW** [comment]
Solved in a similar manner
